src/ParallelAlgorithms/Events/ObserveFields.hpp

```diff
@@ -187,6 +208,19 @@ class ObserveFields<VolumeDim, ObservationValueTag, tmpl::list<Tensors...>,
     const std::string element_name =
         MakeString{} << ElementId<VolumeDim>(array_index) << '/';
 
+    // calculate the interpolation matrices if requested. Else, interp_matrices
+    // will not be default constructed and therefore skipped by applyMatrices.
+    std::array<Matrix, VolumeDim> interp_matrices{};
+    if (interpolation_mesh_) {
+      const auto slices = mesh.slices();
+      const auto interpolation_slices = interpolation_mesh_->slices();
+      for (size_t i = 0; i < VolumeDim; i++) {
+        const auto target_points =
+            Spectral::collocation_points(interpolation_slices.at(i));
+        interp_matrices.at(i) =
+            Spectral::interpolation_matrix(slices.at(i), target_points);
```

Comment:
I suggest to wrap the body of the dimension-loop in another condition that checks `slices.at(i) != interpolation_slices.at(i)`

Reply:
I see it as plausible that the user might specify the default grid as the interpolation grid, so we could add `if (interpolation_mesh_.has_value() && interpolation_mesh.value() != mesh)`.

Otherwise you are addressing the case that the user would request an interpolation to a mesh that is different from the default mesh in certain dimensions only. Is this ever going to happen and worth including the condition check?

Reply:
I think it's perfectly possible the mesh only differs in a particular dimension from the simulation mesh. For example if basis and quadrature are the same, we're upsampling to a constant num points and the domain is refined such that some elements have exactly that num points in some dimension.